### PR TITLE
Add online chess

### PR DESCRIPTION
**Pull request by matt989253**

### CODE REVIEW

File: mobile/src/main/java/ch/epfl/sdp/mobile/application/StoreDocuments.kt [hunk 1]
```diff
@@ -46,3 +50,116 @@ fun ProfileDocument?.toProfile(currentUserUid: String?): Profile {
...
+
+/**
+ * A document which represents a game of chess between two users. All the game documents are stored
+ * in the `/games/` collection.
+ *
+ * @param moves
+ * @param whiteId
+ * @param blackId
+ */
+data class ChessDocument(
```
Comment (alexandrepiveteau) [inline mobile/src/main/java/ch/epfl/sdp/mobile/application/StoreDocuments.kt:62]: I'm not sure about this name. Maybe `GameDocument` better reflects what's in the document.

File: mobile/src/main/java/ch/epfl/sdp/mobile/application/StoreDocuments.kt [hunk 2]
```diff
@@ -46,3 +50,116 @@ fun ProfileDocument?.toProfile(currentUserUid: String?): Profile {
...
+
+/**
+ * Deserializes and creates a [Match] from a [ChessDocument]
+ *
+ * For a null [ChessDocument], a default [Match] with
+ * null value and default [Game] positions will be created
+ *
+ * @return The deserialized [Match]
+ */
+fun ChessDocument?.deserialize(): Match {
```
Comment (alexandrepiveteau) [inline mobile/src/main/java/ch/epfl/sdp/mobile/application/StoreDocuments.kt:77]: Great work on algebraic notation support. We should see how we can move this in the `application.chess.notation` package.

File: mobile/src/main/java/ch/epfl/sdp/mobile/application/chess/online/ChessFacade.kt [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  }
+
+  /**
+   * Updates a [Match] in the [Store] Should only be called with a [Match] whose gameId is not null
+   * in order to target the correct document in the [Store]
+   *
+   * @param match The new [Match] to be stores.
+   */
+  suspend fun updateMatch(match: Match) {
+    match.gameId ?: return
```
Comment (alexandrepiveteau) [inline mobile/src/main/java/ch/epfl/sdp/mobile/application/chess/online/ChessFacade.kt:46]: This should probably be a function on a `Match`.

**Review by KurohanaJuri — APPROVED** [review]
Clean code with todo indication to simply the future improvement , Thx !

File: mobile/src/main/java/ch/epfl/sdp/mobile/application/chess/online/ChessFacade.kt [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  /**
+   * Fetches a [Flow] of [List] of [Match]s that a certain [Profile] has going on with any other
+   * player (or even himself)
+   *
+   * @param profile The [Profile] whose [Matches] will be fetched
+   *
+   * @return The [Flow] of [List] of [Match]s for the [Profile]
+   */
+  fun fetchMatchesForUser(profile: Profile): Flow<List<Match>> {
+    val gamesAsWhite = getMatchesForPlayer(colorField = "whiteId", playerId = profile.uid)
```
Comment (alexandrepiveteau) [inline mobile/src/main/java/ch/epfl/sdp/mobile/application/chess/online/ChessFacade.kt:60]: Not sure about this name. Matches are not fetched until the flow is collected, so maybe `matches(forProfile: Profile)` might be better.

File: mobile/src/main/java/ch/epfl/sdp/mobile/state/CompositionLocals.kt [hunk 5]
```diff
@@ -24,10 +28,12 @@ val LocalSocialFacade = compositionLocalOf<SocialFacade> { error("Missing Social
 fun ProvideFacades(
     authentication: AuthenticationFacade,
     social: SocialFacade,
+    chess: ChessFacade,
     content: @Composable () -> Unit,
 ) {
   CompositionLocalProvider(
       LocalAuthenticationFacade provides authentication,
       LocalSocialFacade provides social,
+      LocalChessFacade provides chess,
```
Comment (alexandrepiveteau) [inline mobile/src/main/java/ch/epfl/sdp/mobile/state/CompositionLocals.kt:37]: Missing test for when the CompositionLocal is missing.

**Review by alexandrepiveteau — APPROVED** [review]
Awesome ! This PR still requires some minor refactoring and coverage isn't quite where it should be, but this is a solid start and good enough for now.

Please don't forget to create some issues for the remaining tasks before merging !

File: mobile/src/main/java/ch/epfl/sdp/mobile/application/StoreDocuments.kt (hunk 1 above)
Comment (KurohanaJuri) [reply mobile/src/main/java/ch/epfl/sdp/mobile/application/StoreDocuments.kt:62]: Maybe `matchDocument` ?

**Comment by matt989253** [comment]
All suggestions added to an Issue for next sprint!

**Pull request merged**